### PR TITLE
Feature/change class structure with dict

### PR DESCRIPTION
**Pull request by aslishemesh**
changed the class structure with dictionary
* change all function to support
* changed all tests to support

@caduri plz review.

### CODE REVIEW

File: fox_scraper/asli.log [hunk 1]
```diff
@@ -0,0 +1,56 @@
+“Running nosetests…”
```
Comment (caduri) [inline fox_scraper/asli.log:1]: why is this file in the repo?

File: fox_scraper/fox_scraper.py [hunk 2]
```diff
@@ -46,41 +46,41 @@ def runner(self):
...
+        this function will scrap all main catalog categories
         :return: a list of beutifulSoup objects
         """
         main_categories_content = self.get_parsed_web_content(self.site_address)
         main_categories_links = []
         for link in main_categories_content.find_all(class_='link padding_hf_v '):
             main_categories_links.append(link.get('href'))
-            self.main_categories.append(link.get_text().strip())
-        return main_categories_links
+            self.main_categories_dict.update({str(link.get_text().strip()): link.get('href')})
```
Comment (caduri) [inline fox_scraper/fox_scraper.py]: split this line, use variables and I still don't understand why are you converting string to string

File: fox_scraper/fox_scraper.py [hunk 3]
```diff
@@ -46,41 +46,41 @@ def runner(self):
...
-        self.sub_categories.append([])
         for sub_category in sub_categories_content.find_all('option'):
-            current_sub_category = self.verify_item_encoding((sub_category.get_text()).strip())
-            self.sub_categories[category_num].append(current_sub_category)
-            sub_categories.append(format(sub_category['value']))
-        return sub_categories
+            current_sub_category = sub_category.get_text().strip()
+            current_sub_category = self.verify_item_encoding(current_sub_category)
+            sub_categories.append({"Sub-Category-Name": current_sub_category, "Sub-Category-Link": format(sub_category['value'])})
+        self.sub_categories_dict.update({main_category: sub_categories})
```
Comment (caduri) [inline fox_scraper/fox_scraper.py]: `sub_categories_dict` seems redundant.
You can change `main_categories_dict` to include this array of dicts

File: fox_scraper/fox_scraper.py [hunk 4]
```diff
@@ -94,60 +94,66 @@ def verify_item_encoding(self, item_name):
...
-        :param main_cat_num: main category number
+        :param main_category: main category name
         :return: list of items for this category
         """
         items = []
-        for sub_cat_num in range(len(sub_categories)):
-            cat_sub_category = self.get_parsed_web_content(sub_categories[sub_cat_num])
-            items += self.get_sub_category_catalog(cat_sub_category, main_cat_num, sub_cat_num)
+        for sub_category in self.sub_categories_dict[main_category]:
+            cat_sub_category = self.get_parsed_web_content(sub_category["Sub-Category-Link"])
```
Comment (caduri) [inline fox_scraper/fox_scraper.py]: `sub_category_scraper` would be a better name

File: fox_scraper/fox_scraper.py [hunk 5]
```diff
@@ -94,60 +94,66 @@ def verify_item_encoding(self, item_name):
...
+        :param sub_category: sub-category main category name
         :param cat_sub_category: web content for fox item
-        :param main_cat_num: main category number
-        :return:
+        :return: list of sub-catalog items
         """
         items = []
-        for link in cat_sub_category.find_all(class_='box padding_hf sp_p_padding_hf center border_b border_l margin_b_1'):
-            items.append(self.parse_item(link, main_cat_num, sub_cat_num))
+        for link in cat_sub_category.find_all(
```
Comment (caduri) [inline fox_scraper/fox_scraper.py]: I would separate this also

File: fox_scraper/fox_scraper.py [hunk 6]
```diff
@@ -94,60 +94,66 @@ def verify_item_encoding(self, item_name):
...
     def send_all_items(self, catalog):
+        """
+        send all catalog to rabbitmq
+        :param catalog: list of FoxItem's lists
+        :return:
+        """
+        with FoxSender() as rabbit_sender:
+            for items in catalog:
+                for item in items:
+                    rabbit_sender.send_message(item)
```
Comment (caduri) [inline fox_scraper/fox_scraper.py]: or in one-liner 
`[rabbit_sender.send_message(item) for items in catalog for item in items]`

File: fox_scraper/fox_scraper.py (hunk 3 above)
Comment (caduri) [reply fox_scraper/fox_scraper.py]: next time

**Review by caduri — APPROVED** [review]
LGTM! Just fix the small comments

File: fox_scraper/asli.log (hunk 1 above)
Comment (aslishemesh) [reply fox_scraper/asli.log:1]: deleted in the new feature branch (cron...)

File: fox_scraper/fox_scraper.py (hunk 2 above)
Comment (aslishemesh) [reply fox_scraper/fox_scraper.py]: updated

File: fox_scraper/fox_scraper.py (hunk 4 above)
Comment (aslishemesh) [reply fox_scraper/fox_scraper.py]: updated

File: fox_scraper/fox_scraper.py (hunk 5 above)
Comment (aslishemesh) [reply fox_scraper/fox_scraper.py]: updated

**Pull request closed without merging**